src/main/java/eu/dissco/backend/component/FdoComponent.java

```diff
@@ -0,0 +1,24 @@
+package eu.dissco.backend.component;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import eu.dissco.backend.properties.FdoProperties;
+import lombok.Getter;
+import org.springframework.stereotype.Component;
+
+@Component
+@Getter
+public class FdoComponent {
+
+  private final JsonNode postRequest;
+
+  public FdoComponent(ObjectMapper mapper, FdoProperties fdoProperties) {
```

Comment:
As this is a Component, you should be able to just inject the mapper

Reply:
I did it this way because otherwise we have to set the post request in a @postConstruct, which is hard to test (would need autowiring/integration testing). I think we also said we wanted to avoid postConstructors in general ([this article ](https://levelup.gitconnected.com/stop-using-postconstruct-in-your-java-applications-2a66fb202cb8)discusses issues with unit testing & postconstruct, among other issues)

Reply:
I don't think so. This is a Component class which means it is a Bean and has access to other Beans. The ObjectMapper is also a Bean and can just be injected here as a private final variable as we do in other places. So we wouldn't need to use a PostConstruct method.

Reply:
The postRequest is not a bean. We could inject the mapper/FdoProperties, but that only works if we use the @RequiredArgsConstructor, so we wouldnt be initializing the postRequest JsonNode during construction. I could make postRequest a bean, or I could remove the @component flag, or I could read a static json from a properties file.

Reply:
Ok yes let's go with this.